File: packages/ts-sdk/test/adapters/runtime/exec.test.ts
import { describe, expect, it } from 'vitest';
import { ExecAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('ExecAdapter', () => {
  it('constructs exec endpoint', async () => {
    const transport = new MockTransport();
    const adapter = new ExecAdapter({ transport });
    await adapter.run('ws', {});
    expect(transport.last()).toMatchObject({ method: 'POST', path: '/api/v1/workspaces/ws/exec' });
  });
});

File: packages/elixir-sdk/test/adapters/runtime/exec_test.exs
defmodule TreeDxSdk.ExecAdapterTest do
  use ExUnit.Case, async: true

  test "run constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Exec.run(client, "ws/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/workspaces/ws%2Fa/exec")
           )
  end
end
